### PR TITLE
feat: Enable new dataset creation flow II

### PR DESCRIPTION
**Pull request by lyndsiWilliams**
### SUMMARY

This PR enables the new create dataset flow in dataset view, the `+ -> Data` menu in the app header, chart creation, and database creation. On the create dataset page (`/dataset/add/`), user is brought to their previous page when the cancel button is clicked and the create chart view when the create dataset button is clicked.

This PR also fixes all the issues introduced in the original iteration of this PR: #22610 

- Issues fixed: 
  - Slow loading navigating to and from SQL Lab
  - Disappearing top nav in SQL Lab
  - Create Dataset menu disabled in top nav's `+` menu
  - Error message occurs when you go to the "create dataset" page after you've deleted a newly connected dataset
  - Already existing datasets are not marked as "already exists" in dataset creation page

### TESTING INSTRUCTIONS

- Check all areas in Superset where user can create a dataset, it should bring you to the new create dataset page: `/dataset/add`
  - All creation spots: `+ Dataset` button in dataset view, the `+ -> Data` menu in the app header, chart creation, and database creation.
  - There should no longer be an Add Dataset modal in any area of the app
- On the new create dataset page, ensure that:
  - Cancel button brings you to the previous page you were on
  - Create Dataset button brings you to the create chart view with the newly created dataset selected

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by codyml — APPROVED** [review]
Looks great!  Left some non-blocking small comments.  Tested the following locally (Docker):
- SQL Lab header no longer missing (https://github.com/apache/superset/pull/22782)
- Deleted database no longer autofilled (https://github.com/apache/superset/pull/22610#issuecomment-1385908014)
- SQL Lab load no longer hangs (https://github.com/apache/superset/pull/22610#issuecomment-1385940520)
- Datasets based on SQLite tables register correctly as already created (https://github.com/apache/superset/pull/22610#issuecomment-1386027957)
- Create dataset button in `RightMenu` no longer disabled when file upload is disabled
- New flow is entered correctly from:
  - "+ Dataset" button in Dataset List view
  - "Create Dataset" menu item in header plus menu
  - "Add a dataset" link on add chart view
  - "Create Dataset" button on connect database success (w/ correct autofilling of DB)
- Dataset is correctly autofilled when chart is created following dataset add
- "Close" returns to previous URL

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Footer/index.tsx [hunk 1]
```diff
@@ -104,13 +100,13 @@ function Footer({
...
           // When a dataset is created the response we get is its ID number
-          goToPreviousUrl();
+          history.push(`/chart/add/?dataset=${datasetObject.table_name}`);
         }
       });
     }
   };
 
-  const CREATE_DATASET_TEXT = t('Create Dataset');
+  const CREATE_DATASET_TEXT = t('Create Dataset and Create Chart');
```
[the remark this answers is not in the document]
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/dataset/AddDataset/Footer/index.tsx]: Good thinking on the text, and good catch on the tooltip in the footer! Fixed in [`this commit`](https://github.com/apache/superset/pull/22835/commits/76b3590962bea5fd457cb8bf790ad811d147c316).

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx [hunk 2]
```diff
@@ -640,6 +642,14 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     onHide();
   };
 
+  const redirectURL = (url: string) => {
```
[the remark this answers is not in the document]
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx:645]: Good idea! Added the comment in [`this commit`](https://github.com/apache/superset/pull/22835/commits/76b3590962bea5fd457cb8bf790ad811d147c316).

**Pull request merged**